userspace/libsinsp/container.h

```diff
@@ -111,7 +111,8 @@ class sinsp_container_manager
 	void identify_category(sinsp_threadinfo *tinfo);
 
 	bool container_exists(const std::string& container_id) const {
-		return m_containers.find(container_id) != m_containers.end();
+		return m_containers.find(container_id) != m_containers.end() ||
+			m_lookups.find(container_id) != m_lookups.end();
```

Comment:
I think I'm confused about when someone should use container_exists, when someone should use should_lookup, and when someone should fetch container info and check m_lookup_state. Non-docker/non-cri seem to use container_exists, docker uses should_lookup, and cri fetches and checks m_lookup_state.